### DISCUSSION TITLE
Ruby 3 support in Docker image?

### DISCUSSION THREAD

**Discussion by ConnorJC3** [renovatebot/renovate#8438] [Request Help] [answered]
I'm using renovate on a repository using ruby 3.0.0. However, whenever renovate runs, it comments an "artifact error" with the following lines:

```
Warning: the running version of Bundler (2.1.4) is older than the version that created the lockfile (2.2.3). We suggest you to upgrade to the version that created the lockfile by running `gem install bundler:2.2.3`.
Could not find gem 'Ruby (~> 3.0.0.0)' in the local ruby installation.
The source contains 'Ruby' at: 2.7.2.137
```

This is doubly annoying because the failing check stops renovate from auto-merging. Is there a way to update the ruby version renovate uses, or disable/ignore this check?

**Comment by rarkins** [comment, marked as answer]
What type of Renovate are you running? e.g. are you using the hosted app, or self-hosted? 

We have Ruby 3.0.0 in our built sidecar images (https://hub.docker.com/r/renovate/ruby/tags?page=1&ordering=last_updated&name=3), so if you are running the app then please create a [minimal reproduction repo](https://github.com/renovatebot/renovate/blob/master/docs/development/minimal-reproductions.md) if the debug logs don't make it obvious what the problem is.

**Reply by ConnorJC3** [reply]
Self-hosted, (container `docker.io/renovate/renovate:24.29.0`) with the following settings:

Renovate environment variables:
```
RENOVATE_TOKEN: <redacted>
RENOVATE_CONFIG_FILE: /opt/renovate/config.json
```

Contents of `/opt/renovate/config.json`
```
{
  "onboardingConfig": { "extends": ["config:base", ":prHourlyLimitNone", ":prConcurrentLimitNone", ":pinVersions", "docker:pinDigests", ":rebaseStalePrs"] },
  "onboardingConfigFileName" : ".renovaterc.json",
  "persistRepoData": true,
  "platform": "github",
  "autodiscover": true,
  "gitAuthor": "Realliance CI <ci@realliance.net>"
}
```

Reproduction:

Example repo: https://github.com/realliance/renovate-repro
Example non-working PR: https://github.com/realliance/renovate-repro/pull/4

Generic steps:
1. Create `.ruby-version` (contents: `ruby-3.0.0`), `Gemfile` (contents: any outdated gem)
2. `bundle install` to create a `Gemfile.lock`, with the following local versions (latest stable)
```
$ ruby -v     
ruby 3.0.0p0 (2020-12-25 revision 95aff21468) [x86_64-linux]

$ bundle -v       
Bundler version 2.2.3
```
3. Create renovate config (either manually or from onboarding PR, doesn't matter)
4. Renovate PR will contain an artifact error

**Reply by rarkins** [reply]
I've upgraded Ruby in our full docker image here: https://github.com/renovatebot/docker-renovate-full/commit/212eff00ad014d231ad6518b00229542550fa68e

However GitHub Actions are down right now so nothing's building

**Reply by ConnorJC3** [reply]
Thanks!
